crates/contracts/solidity/Reader.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import { IStorageAccessible } from "./interfaces/IStorageAccessible.sol";
+
+/// @title A contract for simulating delegate calls on the Settlement contract.
+contract Reader {
```

Comment:
Nit, while the name is very concise, I find it hard to. by itself grasp what this contract is doing. Maybe something like DeployExtensionAndSimulateDelegatecall ?

Reply:
Maybe `SimulateCode` or something? We don't actually deploy (well we do, but that is an implementation detail - if it were able to simulate code in some other way, then we could use that without it affecting the interface).

Reply:
Renamed things a bit (also `s/reader/storage_accessible/g` since the module is really for building calls to `StorageAccessible` contracts) - LMK if the new naming is better or if I should change it again.